tests/test_optimizers_vs_ase.py

```diff
@@ -1,126 +1,278 @@
 import copy
-import functools
+from typing import TYPE_CHECKING, Any
 
+import numpy as np
 import pytest
 import torch
-from ase.filters import FrechetCellFilter
+from ase.build import bulk
+from ase.filters import FrechetCellFilter, UnitCellFilter
 from ase.optimize import FIRE
 
 import torch_sim as ts
-from torch_sim.io import state_to_atoms
+from torch_sim.io import atoms_to_state, state_to_atoms
 from torch_sim.models.mace import MaceModel
-from torch_sim.optimizers import frechet_cell_fire
+from torch_sim.optimizers import frechet_cell_fire, unit_cell_fire
 
 
-try:
+if TYPE_CHECKING:
     from mace.calculators import MACECalculator
-except ImportError:
-    MACECalculator = None
 
 
-@pytest.mark.skipif(MACECalculator is None, reason="MACECalculator not installed")
-def test_torchsim_frechet_cell_fire_vs_ase_mace(
-    rattled_sio2_sim_state: ts.state.SimState,
+@pytest.fixture
+def osn2_sim_state(torchsim_mace_mpa: MaceModel) -> ts.state.SimState:
+    """Provides an initial SimState for rhombohedral OsN2."""
+    # For pymatgen Structure initialization
+    from pymatgen.core import Lattice, Structure
+
+    a = 3.211996
+    lattice = Lattice.from_parameters(a, a, a, 60, 60, 60)
+    species = ["Os", "N"]
+    frac_coords = [[0.75, 0.7501, -0.25], [0, 0, 0]]  # Slightly perturbed
+    structure = Structure(lattice, species, frac_coords, coords_are_cartesian=False)
+    return ts.initialize_state(
+        structure, dtype=torchsim_mace_mpa.dtype, device=torchsim_mace_mpa.device
+    )
+
+
+@pytest.fixture
+def distorted_fcc_al_conventional_sim_state(
```

Comment:
same as above

Reply:
done in [37992f5](https://github.com/Radical-AI/torch-sim/pull/199/commits/37992f5aab69e3b75132006688497dcb1efc671b)